### ISSUE TITLE
bug(E1120): sorcery switch unsupported

### ISSUE THREAD

**Issue by Kristinita** [PyCQA/pylint#2450]
### 1. Summary

Pylint report, that [**switch function of sorcery module**](https://github.com/alexmojaki/sorcery#switch) is incorrect.

### 2. Data

Switch function of sorcery simplifies the syntax of `if/elif/else` blocks.

+ [**Online demonstration on Repl.it**](https://repl.it/@Kristinita/sashapylintsorcerye1120)

+ `sasha__pylint__sorcery_e1120.py`:

```python
"""Docstring."""
from sorcery import switch

for kira_day in range(4, 14, 7):
    print(switch(kira_day, lambda: {
        4: 14,
        11: 4147,
    }))

```

+ Output:

```text
14
4147
```

### 3. Steps to reproduce

```text
pylint sasha__pylint__sorcery_e1120.py
```

### 4. Expected behavior

No warnings and/or errors.

### 5. Actual behavior

```text
D:\SashaDebugging\SashaPython>pylint sasha__pylint__sorcery_e1120.py
************* Module sasha__pylint__sorcery_e1120
sasha__pylint__sorcery_e1120.py:5:10: E1120: No value for argument '_cases' in function call (no-value-for-parameter)

------------------------------------
Your code has been rated at -6.67/10
```

### 6. Environment

```text
D:\SashaDebugging\SashaPython>pylint --version
pylint 2.1.1
astroid 2.0.4
Python 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit (AMD64)]
```

Issue doesn't fix, if `pip install pylint astroid --pre -U`.

Thanks.

**Comment by PCManticore** [comment]
There's a lot of weird stuff going on with [Spell](https://github.com/alexmojaki/sorcery/blob/master/sorcery/core.py#L262) that `pylint` cannot make sense of. Remember that `pylint` is a static analysis tool that needs to parse and interpret your code in order to make sense of it, and it has trouble understanding the use of dynamic features (descriptors, frame manipulation, dynamic changing of scopes, even [decorators](https://github.com/PyCQA/pylint/issues/259)). Given this, I suggest disabling the error in-place for that function, as supporting this use case seems a bit too extreme for a static analysis tool.
